### ISSUE TITLE
(Building for production) Can’t find variable: assert

### ISSUE THREAD

**Issue by lchski** [yoshuawuyts/choo#171]
Using choo to prototype an interface, and I want to build it for production/deployment (supporting IE/Safari is necessary). I followed the suggestions in the readme and landed with this build command:

```
NODE_ENV=production browserify \
    -t envoy \
    -g yo-yoify \
    -g unassertify \
    -g es2020 \
    -g uglifyify \
    index.js \
    | uglifyjs -c -o dist/bundle.js
```

When loading a single view, things work fine. But navigating to another view or performing an action based on `send()` causes an error to appear in the console: `ReferenceError: Can't find variable: assert`.

Looking at `bundle.js`, there seem to be two `assert` statements remaining, even after running `unassertify` as a global transform:

```js
assert.fail("choo: send() cannot be called from Node")
```
and
```js
assert.ifError(o,"nanoraf: infinite loop detected")
```

Admittedly this might not be a choo issue—the one remaining assert is from choo, the other is from nanoraf—and it could very well be `unassertify` that’s causing this mess up, but I wanted to ping folks here first to see if I was going about it wrong somewhere.

**Comment by lchski** [comment]
Okay, I think I might have something:

`unassertify` removes assert() statements following the patterns set out in the [Node.js standard assert API](https://nodejs.org/api/assert.html). (`unassertify` [lists supported patterns](https://github.com/twada/unassertify#supported-patterns).) Looking at the assert statements that remain in the compiled bundle, neither match any of the assert formats:

* [`assert.fail('choo: send() cannot be called from Node');`](https://github.com/yoshuawuyts/choo/blob/master/index.js#L50)
    * Proper/supported format for `assert.fail()`: `assert.fail(actual, expected, message, operator)`
* [`assert.ifError(inRenderingTransaction, 'nanoraf: infinite loop detected')`](https://github.com/yoshuawuyts/nanoraf/blob/master/index.js#L23)
    * Proper/supported format for `assert.ifError()`: `assert.ifError(value)`

This isn’t an issue with `unassertify`, then, but one with these two assert statements specifically. Rewriting the assert statements to use the proper format should fix this issue.

**Comment by lchski** [comment]
Update on this: I created two PRs to address this issue.

* The [one for `nanoraf`](https://github.com/yoshuawuyts/nanoraf/pull/2) alone fixes the specific issue noted here—swapping out upstream `nanoraf` for my fork with the fix causes the compiled JS to work properly, without errors in the console. Looking at the compiled JS, the `nanoraf` call to `assert` is gone.
* The [one for `choo`](https://github.com/yoshuawuyts/choo/pull/173) isn’t necessary to fix the issue noted here—the `nanoraf` PR alone is enough for that. It is necessary, though, to ensure that the `assert` statement is removed from compiled JS.

Let me know if either doesn’t make sense in its implementation; I tried to stick to what made sense and made sure the test suites ran successfully.
